src/datamodel_code_generator/__main__.py

```
@@ -1172,6 +1190,250 @@ def _try_rebuild_model(obj: type) -> None:
+    new_schema["title"] = schema.get("title")
+    new_schema["type"] = "object"
+
+    # Preserve any extra schema properties (like x-python-type)
+    for key in schema:
+        if key not in {"$defs", "properties", "required", "title", "type", "allOf"}:
+            new_schema[key] = schema[key]
+
+    return new_schema
+
+
+def _load_multiple_model_schemas(  # noqa: PLR0912, PLR0914, PLR0915
```

Ruff (0.14.10)

Unused `noqa` directive (non-enabled: `PLR0912`, `PLR0914`, `PLR0915`)

Remove unused `noqa` directive

(RUF100)